databricks/sdk/credentials_provider.py

```diff
@@ -124,11 +176,14 @@ def inner() -> Dict[str, str]:
         token = token_source.token()
         return {'Authorization': f'{token.token_type} {token.access_token}'}
 
-    return inner
+    def token() -> Token:
```

Comment:
The difference from the original is this function. Previous code:
```
return OAuthCredentialsProvider(inner, inner.token)
```
That does not work because `config.py` tries to make a copy of `inner` which uses a "lock" (and cannot be copied). The function works as a wrapper, and allows this object to be copied without copying `inner`.